### ISSUE TITLE
XMLDatabase does not produce parameters for overloaded routines

### ISSUE THREAD

**Issue by elenaem26** [jOOQ/jOOQ#10979]
### Expected behavior
Parameters are generated for overloaded postgreSql procedures pg_catalog. pg_advisory_xact_lock(bigint) and pg_catalog. pg_advisory_xact_lock(integer, integer) when being generated using information_schema.xml
### Actual behavior
Parameters are not generated for overloaded postgres procedures pg_catalog. pg_advisory_xact_lock(bigint) and pg_catalog. pg_advisory_xact_lock(integer, integer) when being generated using information_schema.xml.

### Steps to reproduce the problem
**Code generation configuration:**
```groovy
jooq {
    generateSchemaSourceOnCompilation = false
    xml(sourceSets.main) {
        jdbc {
            driver = 'org.postgresql.Driver'
            url = "URL"
            user = "USER"
            password = "PASSWORD"
        }
        generator {
            name = 'org.jooq.codegen.XMLGenerator'
            database {
                name = 'org.jooq.meta.postgres.PostgresDatabase'
                inputSchema = 'pg_catalog'
                includes = 'pg_catalog.pg_advisory_xact_lock'
            }
            target {
                directory = 'src/main/resources'
            }
        }
    }

    java(sourceSets.main) {
        generator {
            database {
                name = 'org.jooq.meta.xml.XMLDatabase'
                inputSchema = 'pg_catalog'
                includes = 'pg_catalog.pg_advisory_xact_lock'
                properties {
                    property {
                        key = 'dialect'
                        value = 'POSTGRES'
                    }
                    property {
                        key = 'xmlFile'
                        value = 'src/main/resources/information_schema.xml'
                    }
                }
            }
            generate {
                relations = true
                deprecated = false
                records = true
                pojos = true
                pojosEqualsAndHashCode = true
                fluentSetters = true
                javaTimeTypes = true
            }
            target {
                packageName = 'example.jooq'
                directory = 'src/main/java'
            }
        }
    }
}
```

**Steps to reproduce:**
1) Generate xml schema for routines pg_catalog.pg_advisory_xact_lock'. There will be different parameters in the schema that could be distinguished by specific_name tag.
Generated schema:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<information_schema>
    <schemata>
        <schema>
            <catalog_name></catalog_name>
            <schema_name>pg_catalog</schema_name>
            <comment></comment>
        </schema>
    </schemata>
    <routines>
        <routine>
            <specific_catalog></specific_catalog>
            <specific_schema>pg_catalog</specific_schema>
            <specific_name>pg_advisory_xact_lock_1</specific_name>
            <routine_catalog></routine_catalog>
            <routine_schema>pg_catalog</routine_schema>
            <routine_name>pg_advisory_xact_lock</routine_name>
            <routine_type>PROCEDURE</routine_type>
        </routine>
        <routine>
            <specific_catalog></specific_catalog>
            <specific_schema>pg_catalog</specific_schema>
            <specific_name>pg_advisory_xact_lock_2</specific_name>
            <routine_catalog></routine_catalog>
            <routine_schema>pg_catalog</routine_schema>
            <routine_name>pg_advisory_xact_lock</routine_name>
            <routine_type>PROCEDURE</routine_type>
        </routine>
    </routines>
    <parameters>
        <parameter>
            <specific_catalog></specific_catalog>
            <specific_schema>pg_catalog</specific_schema>
            <specific_name>pg_advisory_xact_lock_1</specific_name>
            <ordinal_position>1</ordinal_position>
            <parameter_mode>IN</parameter_mode>
            <parameter_name>_1</parameter_name>
            <data_type>bigint</data_type>
            <character_maximum_length>0</character_maximum_length>
            <numeric_precision>0</numeric_precision>
            <numeric_scale>0</numeric_scale>
            <comment></comment>
        </parameter>
        <parameter>
            <specific_catalog></specific_catalog>
            <specific_schema>pg_catalog</specific_schema>
            <specific_name>pg_advisory_xact_lock_2</specific_name>
            <ordinal_position>1</ordinal_position>
            <parameter_mode>IN</parameter_mode>
            <parameter_name>_1</parameter_name>
            <data_type>integer</data_type>
            <character_maximum_length>0</character_maximum_length>
            <numeric_precision>0</numeric_precision>
            <numeric_scale>0</numeric_scale>
            <comment></comment>
        </parameter>
        <parameter>
            <specific_catalog></specific_catalog>
            <specific_schema>pg_catalog</specific_schema>
            <specific_name>pg_advisory_xact_lock_2</specific_name>
            <ordinal_position>2</ordinal_position>
            <parameter_mode>IN</parameter_mode>
            <parameter_name>_2</parameter_name>
            <data_type>integer</data_type>
            <character_maximum_length>0</character_maximum_length>
            <numeric_precision>0</numeric_precision>
            <numeric_scale>0</numeric_scale>
            <comment></comment>
        </parameter>
    </parameters>
</information_schema>
```
2) Generate java classes using this xml schema.  There will no parameters for procedures at all.
For example:
```java
....
/**
 * This class is generated by jOOQ.
 */
@SuppressWarnings({ "all", "unchecked", "rawtypes" })
public class PgAdvisoryXactLock1 extends AbstractRoutine<java.lang.Void> {

    private static final long serialVersionUID = 1235659372;

    /**
     * Create a new routine call instance
     */
    public PgAdvisoryXactLock1() {
        super("pg_advisory_xact_lock", PgCatalog.PG_CATALOG);
        setOverloaded(true);
    }
}
```

### Versions

- jOOQ: 3.14.3
- Java: 14.0.2
- Database (include vendor): PostgreSQL 12
- OS: MacOs Catalina
- JDBC Driver (include name if inofficial driver): org.postgresql.Driver 42.2.12
- Gradle plugin: nu.studer.jooq 4.2

**Comment by lukaseder** [comment]
Thanks a lot for your report and for your PR. I will look into this shortly

**Comment by lukaseder** [comment]
I can reproduce the issue.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.4 (#10985). Thanks again for your detailed report!
